crates/tx5/src/backend.rs

```diff
@@ -0,0 +1,134 @@
+//! Backend modules usable by tx5.
+
+use std::io::Result;
+use std::sync::Arc;
+
+use futures::future::BoxFuture;
+
+use crate::{Config, PubKey};
+use tx5_core::deps::serde_json;
+
+#[cfg(feature = "backend-go-pion")]
+mod go_pion;
+
+/// Backend modules usable by tx5.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum BackendModule {
+    #[cfg(feature = "backend-go-pion")]
+    /// The Go Pion-based backend.
+    GoPion,
+
+    #[cfg(feature = "backend-webrtc-rs")]
+    /// The Webrtc-RS-based backend.
+    WebrtcRs,
+
+    /// The mock backend.
+    Mock,
+}
+
+impl Default for BackendModule {
+    #[allow(unreachable_code)]
+    fn default() -> Self {
+        #[cfg(feature = "backend-go-pion")]
+        return Self::GoPion;
+        #[cfg(feature = "backend-webrtc-rs")]
+        return Self::WebrtcRs;
+        Self::Mock
+    }
+}
```

Comment:
My hope was to make the backend features non-exclusive, hence the fallback returns.

Reply:
What happens in Callums suggestion when you compile with both features? Compiler error?

Reply:
Yeah it would be a compiler error for multiple defined defaults. You could fix it with more `cfg_attr` like setting `WebrtcRs` to only default if `go-pion` isn't enabled:

```rust
#[cfg(feature = "backend-webrtc-rs")]
#[cfg_attr(not(feature = "backend-go-pion"), default)]
/// The Webrtc-RS-based backend.
WebrtcRs,
```

but I can imagine that this starts to look unruly as we possibly add more backends.